src/test/java/com/VetApp/PurrgentCare/services/AccountServiceImplementationTest.java

```diff
@@ -51,127 +51,134 @@ public void setup() {
+        final var actual = serviceUnderTest.addAccount(fakeAccountRequest);
 
         // then
-        then(actual).isEqualTo(expected);
+        verify(mockAccountRepository, times(1)).save(fakeAccount);
+        then(actual).isEqualTo(fakeAccountResponse);
     }
 
+    @Test
+    public void deleteAccount_whenValidInput() {
+        // given
+        final var fakeAccountId = fakeDataGenerator.generateRandomInteger();
 
-    // attempting to create tests for updating accounts but getting
-    // stuck because we do not have the ability to set accountHolders & pets
-    // --Unsure of how to continue as of now.
+        // when
+        serviceUnderTest.deleteAccount(fakeAccountId);
+
+        // then
+        verify(mockAccountRepository, times(1)).deleteById(fakeAccountId);
+
+    }
 
 
     @Test
     public void updateAccount_whenAccountExists_returnsUpdatedAccount() {
         // given
+        final var fakeAccountRequest = fakeDataGenerator.generateFakeAccountRequest();
         final var fakeAccountId = fakeDataGenerator.generateRandomInteger();
-        final var fakeActiveTrue = Boolean.TRUE;
-        final var fakeActiveFalse = Boolean.FALSE;
-        final var fakeOriginalDateCreated = fakeDataGenerator.generateRandomDate();
-        final var fakeUpdatedDateCreated = fakeDataGenerator.generateRandomDate();
-        final var fakePetList = fakeDataGenerator.generateDefaultPetList();
-        final var fakePeopleList = fakeDataGenerator.generatePersonList(fakeDataGenerator.generateRandomInteger());
-
-        final var fakeOriginalAccount = fakeDataGenerator.generateAccount(fakeAccountId, fakeActiveTrue, fakeOriginalDateCreated, fakePetList, fakePeopleList);
-        final var fakeUpdatedAccount = fakeDataGenerator.generateAccount(fakeAccountId, fakeActiveFalse, fakeUpdatedDateCreated, fakePetList, fakePeopleList);
+        final var fakeOriginalAccount = fakeDataGenerator.generateDefaultAccount();
+        final var fakeUpdatedAccount = fakeDataGenerator.generateDefaultAccount();
+        fakeUpdatedAccount.setId(fakeOriginalAccount.getId());
+        final var fakeAccountResponse = fakeDataGenerator.generateDefaultAccountResponse();
+        given(mockMapper.map(fakeAccountRequest, Account.class)).willReturn(fakeUpdatedAccount);
         given(mockAccountRepository.findById(fakeAccountId)).willReturn(Optional.of(fakeOriginalAccount));
-        when(mockAccountRepository.save(fakeOriginalAccount)).thenReturn(fakeUpdatedAccount);
+        given(mockAccountRepository.save(fakeOriginalAccount)).willReturn(fakeUpdatedAccount);
+        given(mockMapper.map(fakeUpdatedAccount, AccountResponse.class)).willReturn(fakeAccountResponse);
 
         // when
-        final var actual = serviceUnderTest.updateAccount(fakeUpdatedAccount, fakeAccountId);
+        final var actual = serviceUnderTest.updateAccount(fakeAccountRequest, fakeAccountId);
+        verify(mockAccountRepository).save(accountCaptor.capture());
```

Comment:
verify should be in the then portion of the test b/c it's a validation step.

Reply:
The test fails if I move that verify after line 163.
Account capturedAccount = accountCaptor.getValue();
"No argument value was captured!"

This captor structure came from phind, so I don't have a solid understanding of it.